### ISSUE TITLE
Bug: `typescript:install` silently fails to register the ServiceProvider but reports success

### ISSUE THREAD

**Issue by christophstockinger** [spatie/laravel-typescript-transformer#86]
**Package:** `spatie/laravel-typescript-transformer`  
**Version:** `3.2.0`  
**Laravel:** `13.x` | **PHP:** `8.4`

---

### What happens

Running `php artisan typescript:install` prints **"TypeScript Transformer Service Provider registered."** — but the provider is **never actually added** to `bootstrap/providers.php`. Running `php artisan typescript:transform` afterwards fails with:

Please, first publish the TypeScriptTransformerServiceProvider and configure it.

---

### Terminal proof

```bash
$ php artisan typescript:transform
Please, first publish the TypeScriptTransformerServiceProvider and configure it.

$ php artisan typescript:install
Publishing TypeScript Transformer Service Provider...
TypeScript Transformer Service Provider already installed.
TypeScript Transformer Service Provider registered.   ← false positive

$ php artisan typescript:transform
Please, first publish the TypeScriptTransformerServiceProvider and configure it.   ← still broken
```

---

### Root cause

`InstallTypeScriptTransformerCommand::registerServiceProvider()` uses `str_replace` to inject the provider into `bootstrap/providers.php`, searching for the fully-qualified class name:

```php
// src/Commands/InstallTypeScriptTransformerCommand.php
"{$namespace}\\Providers\AppServiceProvider::class,".PHP_EOL,
```

However, bootstrap/providers.php in many standard Laravel projects uses short class names with use imports at the top — not fully-qualified names:

```php
// bootstrap/providers.php (as generated by Laravel)
use App\Providers\AppServiceProvider;

return [
    AppServiceProvider::class,  // ← no FQCN, str_replace never matches
];
```

The str_replace returns the original string unchanged. file_put_contents writes it back silently. The command still reports success.

---

###  Expected behavior

Either:
- The command handles both FQCN and short-name formats (e.g. via regex), or
- It fails with a clear error when registration was not possible, instead of printing a false success message

---

### Workaround

Manually add the provider to bootstrap/providers.php:

```php
return [
    App\Providers\TypeScriptTransformerServiceProvider::class,
    // ...
];
```

---

###  Additional issues

1. The error message from typescript:transform says "first publish the ServiceProvider" — which is misleading, because the provider file exists. The actual problem is that it is not registered. A clearer message would help users diagnose the issue without digging into the source code.
2. The Laravel Data integration section of the v3 documentation appears to be outdated and does not reflect the current API.
